aten/src/THC/THCCachingAllocator.cpp

```diff
@@ -516,86 +519,86 @@ static void CudaCachingDeleter(void* ptr) {
 // NB: I decided not to fold this into THCCachingAllocator, because the latter
 // has a lot more methods and it wasn't altogether clear that they should
 // actually be publically exposed
-struct CudaCachingAllocator : public at::Allocator {
-  at::DataPtr allocate(size_t size) const override {
+struct CudaCachingAllocator : public Allocator {
+  DataPtr allocate(size_t size) const override {
     int device;
-    THCudaCheck(cudaGetDevice(&device));
+    C10_CUDA_CHECK(cudaGetDevice(&device));
     void* r = nullptr;
     if (size != 0) {
-      caching_allocator.malloc(&r, size, at::cuda::getCurrentCUDAStream(device));
+      caching_allocator.malloc(&r, size, cuda::getCurrentCUDAStream(device));
     }
-    return {r, r, &CudaCachingDeleter, at::Device(at::DeviceType::CUDA, device)};
+    return {r, r, &CudaCachingDeleter, Device(DeviceType::CUDA, device)};
   }
-  at::DeleterFnPtr raw_deleter() const override {
+  DeleterFnPtr raw_deleter() const override {
     return &CudaCachingDeleter;
   }
 };
 
 CudaCachingAllocator device_allocator;
 
-THC_API at::Allocator* THCCachingAllocator_get(void)
+Allocator* THCCachingAllocator_get(void)
```

Comment:
Removing all these `THC_API` macros is a change you don't mention in the description. Is this intentional? Are you sure this shouldn't be `C10_API` or something like that?

Reply:
It's intentional. There's no point in putting `_API` macros in a cpp file; the previous declarations were in error.